tgui/packages/tgui/interfaces/StripMenu.tsx

```diff
@@ -0,0 +1,394 @@
+                          'vertical-align': 'middle',
+                        }}
+                      />
+                    );
+
+                    tooltip = item.name;
+                  } else if ('obscured' in item) {
+                    content = (
+                      <Icon
+                        name={
+                          item.obscured === ObscuringLevel.Completely
+                            ? 'ban'
+                            : 'eye-slash'
+                        }
+                        size={3}
+                        ml={0}
+                        mt={1.3}
+                        style={{
+                          'text-align': 'center',
+                          height: '100%',
+                          width: '100%',
+                        }}
+                      />
+                    );
+
+                    tooltip = `obscured ${slot.displayName}`;
+                  }
+
+                  return (
+                    <Stack.Item
+                      key={key}
+                      style={{
+                        width: BUTTON_DIMENSIONS,
+                        height: BUTTON_DIMENSIONS,
+                      }}>
+                      <Box
+                        style={{
+                          position: 'relative',
+                          width: '100%',
+                          height: '100%',
+                        }}>
+                        <Button
+                          onClick={() => {
+                            act('use', {
+                              key: keyAtSpot,
+                            });
+                          }}
+                          fluid
+                          tooltip={tooltip}
+                          style={{
+                            background: item?.interacting
+                              ? 'hsl(39, 73%, 30%)'
+                              : undefined,
+                            position: 'relative',
+                            width: '100%',
+                            height: '100%',
+                            padding: 0,
+                          }}>
+                          {slot.image && (
+                            <Box
+                              as="img"
+                              src={resolveAsset(slot.image)}
+                              opacity={0.7}
+                              style={{
+                                position: 'absolute',
+                                width: '32px',
+                                height: '32px',
+                                left: '50%',
+                                top: '50%',
+                                transform:
+                                  'translateX(-50%) translateY(-50%) scale(0.8)',
+                              }}
+                            />
+                          )}
+
+                          <Box style={{ position: 'relative' }}>{content}</Box>
```

Comment:
Rather than determining content as above - can the content here just be a component that passes in the item? The existing approach would just re-render every box every tick. By putting this in its own component it will only rerender if changed